src/server/main_service.cc

```diff
@@ -348,11 +348,12 @@ class EvalSerializer : public ObjectExplorer {
   }
 
   void OnDouble(double d) final {
-    if (rb_->IsResp3() || !GetFlag(FLAGS_lua_resp2_legacy_float)) {
-      rb_->SendDouble(d);
-    } else {
-      long val = d >= 0 ? static_cast<long>(floor(d)) : static_cast<long>(ceil(d));
+    thread_local const bool send_double_as_long = GetFlag(FLAGS_lua_resp2_legacy_float);
```

Comment:
What's the point of thread local here ? Also I think you inverted the condition, now you sent `long` on resp 2`

Reply:
`thread_local` is to not compute the flag each time, as I see this cannot be reset by `config set`. It will only be computed the first time.

I didn't intend to invert the condition, but it looks like it changes behavior for RESP2. Earlier it was:

* if resp3 OR flag is false, send double
* otherwise send long

I changed it to
* if flag is true, send long
* otherwise send double

It should be:
* if flag is true OR resp2, send long
* otherwise send double

I will change it for resp2

Reply:
`thread_local` variables are computed once for the thread lifetime so the end result should be the same. I will change it to static though as I think `thread_local` doesn't make much sense here as this is `main_service` and nothing per shard and static is easier to understand in this context.

I see a bug is introduced by my changes, I realize by looking at the `EvalResp` test. Will change this code.

Reply:
uh implicit static for thread locals :cry:

Reply:
you do not need `thread_local` here for sure. we use thread locals when we mutate the setting via `CONFIG SET`.
you can just use a global static.

Reply:
I thought I had introduced some unwanted change to behavior with respect to RESP2, but on review it seems correct to me. 

Earlier, if the flag was _not_ set, or we had resp3, we sent double. This means the flag did not work for resp3.

Now if we have the flag set, we send long (irrespective of resp2/resp3). If it is not set, we send double, again irrespective of the resp version. 

This behavior seems correct to me. A unit test fails when run with others (not in isolation), because of the static evaluation of the flag, it is not reevaluated even if set in the `EvalBug2664`